packages/components/src/placeholder/test/index.tsx

```diff
@@ -0,0 +1,157 @@
+/**
+ * External dependencies
+ */
+import { render, screen, within } from '@testing-library/react';
+
+/**
+ * WordPress dependencies
+ */
+import { more } from '@wordpress/icons';
+import { useResizeObserver } from '@wordpress/compose';
+
+/**
+ * Internal dependencies
+ */
+import BasePlaceholder from '../';
+import type { WordPressComponentProps } from '../../ui/context';
+import type { PlaceholderProps } from '../types';
+
+jest.mock( '@wordpress/compose', () => {
+	return {
+		...jest.requireActual( '@wordpress/compose' ),
+		useResizeObserver: jest.fn( () => [] ),
+	};
+} );
+
+const Placeholder = (
+	props: Omit<
+		WordPressComponentProps< PlaceholderProps< unknown >, 'div', false >,
+		'ref'
+	>
+) => <BasePlaceholder data-testid="placeholder" { ...props } />;
+
+const getPlaceholder = () => screen.getByTestId( 'placeholder' );
+
+const getLabel = () => {
+	const placeholder = getPlaceholder();
+	return placeholder.querySelector( '.components-placeholder__label' );
+};
```

Comment:
I don't _love_ this... I feel like the label should have some kind of meaningful semantics and not be a stray piece of text 🤔 But that's more of a problem with the component itself, so we'll have to leave it be for the purpose of this PR!

Reply:
Should we add a comment about it, at least?

Reply:
👍 

Updated the test to only use `querySelector` when we check if the elements is empty/don't exists. Added a comment on the where we use `querySelector`.